parachains/pallets/collective-content/src/lib.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (C) 2023 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Managed collective content.
+//!
+//! The pallet provides the functionality to store different types of the content.
+//! The content presented as a [Cid] of the IPFS document which might contain any type of data.
+//! Every type of the content has its own origin to be manage. The origins are configurable by clients.
+//!
+//! Content types:
+//! - the collective [charter](pallet::Charter). A single document managed by [CharterOrigin](pallet::Config::CharterOrigin).
+//! - the collective [announcements](pallet::Announcements). A list of announcements managed by [AnnouncementOrigin](pallet::Config::AnnouncementOrigin).
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+pub mod weights;
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+use frame_support::BoundedVec;
+use sp_core::ConstU32;
+
+/// IPFS compatible CID.
+// worst case 2 bytes base and codec, 2 bytes hash type and size, 64 bytes hash digest.
+pub type Cid = BoundedVec<u8, ConstU32<68>>;
```

Comment:
Ive seen few implementations of the CID in or repos (at least two). 
Here I keep it simple instead introducing new. I believe we need to agree first on one type for it, and keep it consistent.

Reply:
Storing CID as a bounded vec sounds good to me

Reply:
We already have a type for cid, maybe we should push it down to a more common place:
https://github.com/paritytech/substrate/blob/8cfe326e4e33c5077fc67f197d6a13dd871881c7/frame/alliance/src/types.rs#L69
We shouldn't have multiple different types for cids.

Reply:
The alternative is we could call this type an OpaqueCid maybe?